validator/client/beacon-api/state_validators.go

```diff
@@ -86,12 +85,19 @@ func (c beaconApiStateValidatorsProvider) getStateValidatorsHelper(
 		}
 	}
 
-	reqBytes, err := json.Marshal(req)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to marshal request into JSON")
+	queryParams := url.Values{}
+	for _, id := range req.Ids {
+		queryParams.Add("id", id)
+	}
+	for _, st := range req.Statuses {
+		queryParams.Add("status", st)
 	}
+
+	query := buildURL(endpoint, queryParams)
 	stateValidatorsJson := &beacon.GetValidatorsResponse{}
-	if err = c.jsonRestHandler.Post(ctx, endpoint, nil, bytes.NewBuffer(reqBytes), stateValidatorsJson); err != nil {
+
+	err := c.jsonRestHandler.Get(ctx, query, stateValidatorsJson)
```

Comment:
Do not change this to `GET`. The `POST` version supports more validator indices because the length of a query string is limited.

Reply:
I changed this to `GET` as we have GET endpoints only to get validator(s) data in the beacon api [spec](https://ethereum.github.io/beacon-APIs/#/Beacon/getStateValidators).